src/controllers/sheltersController.js

```diff
@@ -14,6 +16,27 @@ const getShelters = async (req, res, next) => {
 		});
 };
 
+const postShelters = async (req, res, next) => {
+
+	//check all attrs are provided (input validation)
+	var attrs = ["address","emailAddress","password","phoneNumber","shelterName"];
+	if(!inputValidation.hasAllAttrs(req.body, attrs)) res.status(400).send("Please provide required input attributes");
+	else if(inputValidation.includesNullorEmpty(req.body, ["website"])) res.status(400).send("Please provide required input values");
```

Comment:
I am confused how this is different than the previous line. Is it just verifying "website" isn't just a blank string?

Reply:
`hasAllAttrs `checks if the backend received **all the attributes** from the client of not. For example, if the client only sends `["emailAddress","password"]`, it will return an error.
`includesNullorEmpty` checks null values. For example, the client sends all the attributes but `phoneNumber ` is null. In that case, this function will return an error. However, `website` is nullable in the database...so we pass `website` in the  "exceptions" parameter. Let me know if you have questions!

Reply:
Neat! Totally makes sense, thanks for the explanation